Move MFA account setup to snippet and include in various docs

edwardchalstrey1 opened the pull request:
## :white_check_mark: Checklist

<!--
Thank you for your pull request!

Before going any further, please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.
Replace the empty checkboxes [ ] below with checked ones [x] accordingly.
-->

- [x] You have given your pull request a meaningful title (_e.g._ `Enable foobar integration` rather than `515 foobar`).
- [x] You are targeting the **develop branch**.
- [x] Your branch is up-to-date with the **develop branch** (you probably started your branch from `develop` but it may have changed since then).
- [x] If-and-only-if your changes are not yet ready to merge, you have marked this pull request as a **draft** pull request and added '[WIP]' to the title.
- [x] If-and-only-if you have changed any Powershell code, you have run the code formatter. You can do this with `./tests/AutoFormat_Powershell.ps1 -TargetPath <path to file or directory>`.

### :arrow_heading_up: Summary

Moves MFA account setup instructions into a new snippet and links to it in the following:

1. [SHM deploy doc set up admin account](https://alan-turing-institute.github.io/data-safe-haven/v4.0.0/deployment/deploy_shm.html#activate-and-configure-your-new-internal-admin-account)
2. [SRE deploy doc set up non-privileged user account](https://alan-turing-institute.github.io/data-safe-haven/v4.0.0/deployment/deploy_sre_microsoft_rds.html#verify-non-privileged-user-account-is-set-up)
3. [User guide set up your account](https://alan-turing-institute.github.io/data-safe-haven/v4.0.0/roles/researcher/user_guide_msrds.html#set-up-your-account)

### :closed_umbrella: Related issues

Closes #1305

### :microscope: Tests

- [x] Test the docs look good by building locally

edwardchalstrey1 committed:
move MFA instructions into snippet

edwardchalstrey1 committed:
link to snippet in deploy shm doc

edwardchalstrey1 committed:
add MFA snippet to non-privileged user account setup

jemrobinson reviewed (approved):
LGTM. Have you checked that the documentation builds correctly locally? If so, feel free to merge this.

edwardchalstrey1 commented:
> LGTM. Have you checked that the documentation builds correctly locally? If so, feel free to merge this.

nope @jemrobinson hence draft - how do I do it, with Sphinx?

edwardchalstrey1 committed:
rename for consistency

jemrobinson commented:
> > LGTM. Have you checked that the documentation builds correctly locally? If so, feel free to merge this.
> 
> nope @jemrobinson hence draft - how do I do it, with Sphinx?

```bash
> pip install -r docs/build/requirements.txt
> make -C docs html
```

this should build the docs for your current branch in `docs/_output`.

edwardchalstrey1 commented:
> > > LGTM. Have you checked that the documentation builds correctly locally? If so, feel free to merge this.
> > 
> > 
> > nope @jemrobinson hence draft - how do I do it, with Sphinx?
> 
> ```shell
> > pip install -r docs/build/requirements.txt
> > make -C docs html
> ```
> 
> this should build the docs for your current branch in `docs/_output`.

Works, but I had to edit the `conf.py` when building from my fork, because these lines don't work as my fork has no releases:

```
repo_name = repo.remotes.origin.url.split(".git")[0].split("/")[-1]
releases = sorted((t.name for t in repo.tags), reverse=True)
```

jemrobinson commented:
You should be able to pull the tags in with something like:

```
git fetch upstream --tags # to get the upstream tags (assuming you've added the original repo as 'upstream')
git push -f origin develop # to push the tags to your fork
```

See here: https://stackoverflow.com/questions/70678073/how-do-i-sync-tags-to-a-forked-github-repo

edwardchalstrey1 commented:
> You should be able to pull the tags in with something like:
> 
> ```
> git fetch upstream --tags # to get the upstream tags (assuming you've added the original repo as 'upstream')
> git push -f origin develop # to push the tags to your fork
> ```
> 
> See here: https://stackoverflow.com/questions/70678073/how-do-i-sync-tags-to-a-forked-github-repo

I did:

```
git remote add upstream https://github.com/alan-turing-institute/data-safe-haven
git fetch --tags upstream
git push --tags
```

worked for me

edwardchalstrey1 committed:
fix link

edwardchalstrey1 committed:
deleted commented

edwardchalstrey1 committed:
move snippet link into user guide

edwardchalstrey1 committed:
move snippet link into SRE deploy docs

edwardchalstrey1 committed:
move mfa snippet

edwardchalstrey1 committed:
renumber MFA snippet

edwardchalstrey1 committed:
dont use <summary><details> block in SRE doc

edwardchalstrey1 committed:
singe section for Ensure non-privileged user account has MFA enabled

jemrobinson commented:
> I did:
> 
> ```
> git remote add upstream https://github.com/alan-turing-institute/data-safe-haven
> git fetch --tags upstream
> git push --tags
> ```
> 
> worked for me

Maybe you can add this to the bit of the CONTRIBUTING docs that tells you to fork the repo?

edwardchalstrey1 commented:
hmm, this is fiddly because although the snippet can be incorporated from different locations in the file structure, it's difficult to link to the same images (for MFA screenshots) since different docs have different subdir levels- perhaps there is some way by changing the `:relative-images:` option, will have to investigate further

edwardchalstrey1 committed:
Merge branch 'develop' into mfa-snippet

edwardchalstrey1 committed:
add :relative-images:

edwardchalstrey1 committed:
duplicate images and add warning (hack fix)

edwardchalstrey1 committed:
add spaces

edwardchalstrey1 committed:
add end space

edwardchalstrey1 committed:
remove duplicate snippet

edwardchalstrey1 committed:
reformat lists

edwardchalstrey1 committed:
header level consistent

jemrobinson commented:
Why are all the images showing up as new? Are they duplicated or moved from somewhere else?

jemrobinson reviewed (changes requested):
If the snippet images don't work, maybe it's easier to just keep a single non-snippet version in the user guide and link to it from `deploy_shm.md`?

edwardchalstrey1 commented:
> Why are all the images showing up as new? Are they duplicated or moved from somewhere else?

@jemrobinson unfortunately I think the only solution is to duplicate them, which is unfortunate but although I tried [this suggestion](https://myst-parser.readthedocs.io/en/v0.13.5/using/howto.html?highlight=relative-images#include-a-file-from-outside-the-docs-folder-like-readme-md) in the myst docs, it doesn't seem to work. The nice thing is it's still a single snippet, it only becomes an issue if we want to replace any of the images, but I left some comments in the markdown to explain that

edwardchalstrey1 commented:
> If the snippet images don't work, maybe it's easier to just keep a single non-snippet version in the user guide and link to it from `deploy_shm.md`?

I guess that would work too! But this way we save people flicking back and forth between docs at least

edwardchalstrey1 commented:
Maybe it should even be a separate page altogether rather than a section and just link to that? Might make it easier to find

jemrobinson commented:
> Maybe it should even be a separate page altogether rather than a section and just link to that? Might make it easier to find

Yes. I think the main reason for not doing that is that we want to generate a single PDF for the user guide. We could always reconsider that though.

edwardchalstrey1 commented:
> > Maybe it should even be a separate page altogether rather than a section and just link to that? Might make it easier to find
> 
> Yes. I think the main reason for not doing that is that we want to generate a single PDF for the user guide. We could always reconsider that though.

@jemrobinson so which of the below would you prefer?

- [ ] Keep the snippets working as in this PR (but caveat of duplicate images for screenshots)
- [x] Just have the account creation/MFA instructions on one doc (e.g. the user guide) and link to that from the others (SRE and SHM deployment docs), not a snippet
- [ ] Entirely separate doc page for account creation/MFA. Linked to from all 3 docs where this needs to be followed.

jemrobinson commented:
> Just have the account creation/MFA instructions on one doc (e.g. the user guide) and link to that from the others (SRE and SHM deployment docs), not a snippet

I think this one is the best solution for the moment, thanks :)

edwardchalstrey1 committed:
link to user guide instead of including snippet in deploy docs

edwardchalstrey1 committed:
remove duplicate images

edwardchalstrey1 committed:
remove commented warning

edwardchalstrey1 committed:
pass last commit lint

jemrobinson commented on docs/deployment/deploy_shm.md:
```diff
@@ -358,20 +358,10 @@ In the next step we will delete the external admin account created for the user
-- Enter your username (`aad.admin.firstname.lastname@<SHM domain>`)and the new password
-- Click `Next` at the `Help us to protect your account` prompt
-- Follow the instructions to configure `Microsoft Authenticator`
+The administrators you have just set up can activate their accounts by following the same steps as in the {ref}`user guide <user_setup>`.
```
Can we link to the section here rather than just the document?

edwardchalstrey1 replied:
It does but I can make it more clear

jemrobinson replied:
Ah OK. No problem then :).

jemrobinson commented on docs/roles/researcher/user_guide_msrds.md:
````diff
@@ -10,6 +10,14 @@
 
+(user_setup)=
+
+## {{closed_lock_with_key}} Set a password
+
+```{include} snippets/13_MFA.partial.md
+:relative-images:
+```
````
This feels like it covers more than "Set a password". Can you think of a better name for this section?

edwardchalstrey1 committed:
rename section header

edwardchalstrey1 committed:
update references to password and  mfa setup
